File: seava.mod.ad.presenter/src/main/java/seava/ad/presenter/impl/system/model/DataSourceRpcLov_Ds.java
/**
 * Solutii Ecommerce, Automatizare, Validare si Analiza | Seava.ro
 * Copyright: 2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package seava.ad.presenter.impl.system.model;

import seava.ad.domain.impl.system.DataSourceRpc;
import seava.j4e.api.annotation.Ds;
import seava.j4e.api.annotation.DsField;
import seava.j4e.api.annotation.SortField;
import seava.j4e.presenter.impl.model.AbstractTypeNTLov_Ds;

@Ds(entity = DataSourceRpc.class, sort = {@SortField(field = DataSourceRpcLov_Ds.f_name)})
public class DataSourceRpcLov_Ds extends AbstractTypeNTLov_Ds<DataSourceRpc> {

	public static final String ALIAS = "ad_DataSourceRpcLov_Ds";

	public static final String f_dataSourceId = "dataSourceId";
	public static final String f_dataSourceName = "dataSourceName";

	@DsField(join = "left", path = "dataSource.id")
	private String dataSourceId;

	@DsField(join = "left", path = "dataSource.name")
	private String dataSourceName;

	public DataSourceRpcLov_Ds() {
		super();
	}

	public DataSourceRpcLov_Ds(DataSourceRpc e) {
		super(e);
	}

	public String getDataSourceId() {
		return this.dataSourceId;
	}

	public void setDataSourceId(String dataSourceId) {
		this.dataSourceId = dataSourceId;
	}

	public String getDataSourceName() {
		return this.dataSourceName;
	}

	public void setDataSourceName(String dataSourceName) {
		this.dataSourceName = dataSourceName;
	}
}

File: seava.mod.ad.ui.extjs/src/main/resources/webapp/js/ds/DataSourceRpcLov_Ds.js
/**
 * Solutii Ecommerce, Automatizare, Validare si Analiza | Seava.ro
 * Copyright: 2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
	
Ext.define(Main.ns.ad + "DataSourceRpcLov_Ds", {
	extend: 'Ext.data.Model',
	
	fields: [
		{name:"dataSourceId", type:"string"},
		{name:"dataSourceName", type:"string"},
		{name:"id", type:"string"},
		{name:"name", type:"string"},
		{name:"description", type:"string"},
		{name:"active", type:"boolean"},
		{name:"refid", type:"string"}
	]
});
